lib/bitex/payment.rb

```diff
@@ -3,23 +3,24 @@ module Bitex
   # Documentation here!
   #
   class Payment
-    attr_accessor :id, :user_id, :amount, :currency_id, :expected_quantity,
-      :previous_expected_quantity, :confirmed_quantity, :unconfirmed_quantity,
-      :valid_until, :quote_valid_until, :last_quoted_on, :status, :address,
-      :settlement_currency_id, :settlement_amount, :keep, :merchant_reference,
-      :customer_reference
+    attr_accessor :id, :user_id, :amount, :currency_id, :expected_quantity, :previous_expected_quantity, :confirmed_quantity,
+                  :unconfirmed_quantity, :valid_until, :quote_valid_until, :last_quoted_on, :status, :address,
+                  :settlement_currency_id, :settlement_amount, :keep, :merchant_reference, :customer_reference
 
     # @visibility private
     def self.from_json(json)
       new.tap do |thing|
-        json.each do |key,raw_value|
+        json.each do |key, raw_value|
           next if raw_value.nil?
 
-          value = if %i[valid_until quote_valid_until last_quoted_on].include?(key.to_sym)
-            Time.at(raw_value)
-          else
-            raw_value
-          end
+          value =
+            if [:valid_until, :quote_valid_until, :last_quoted_on].include?(key.to_sym)
+              Time.at(raw_value)
+            else
+              raw_value
+            end
+
+          thing.send("#{key}=", value) rescue nil
```

Comment:
Please do a more thorough review of the diff before committing. This is undoing some of your own changes from PR #8.